AI Integration: describe types (#7219)
DevExpress/devextreme-documentation: api-reference/40 Common Types/AIIntegration/AIIntegration.md

---
id: AIIntegration
module: common/ai-integration
export: AIIntegration
---
---
##### shortDescription

> A type of the main AI integration class.

Reviewer: "Type" and "Class" are synonymous in my mind. If I write something  like this: 

var myButton = new Button(); 

Then "Button" is a class and it is a type. 

So if I were to simply edit this description, I'd say something like this: 

```suggestion
The main AI integration class.
```

But that doesn't really say much... We probably need to indicate that this is our custom class designed for our custom components/widgets. 

```suggestion
A class that activates AI services in DevExpress UI components. 
```

---

> You must create an instance of this class by passing in an object of the [AIProvider](/Documentation/ApiReference/Common_Types/AIProvider/) type. This instance should then be passed to any components that support AI features.

Reviewer: ```suggestion
This object's constructor accepts an [AIProvider](/Documentation/ApiReference/Common_Types/AIProvider/) object that specifies AI service settings. Pass the created `AIIntegration` object to components where you want to activate AI capabilities.
```

---
##### jQuery

    <!-- tab: index.js -->
    const aiIntegration = new DevExpress.aiIntegration(provider);

    $("#htmlEditor").dxHtmlEditor({
        // ...
        aiIntegration,
    });

##### Angular

    <!-- tab: app.component.ts -->

>     import { AIIntegration } from 'devextreme-angular/ai-integration';

Reviewer 2: Please fix for all samples in frameworks
```suggestion
    import { AIIntegration } from 'devextreme-angular/common/ai-integration';
```

    // ...
    export class AppComponent {

>         ai = new AIIntegration(provider);

Reviewer 3: It's better not to use “ai”, I think. It might confuse users. Don't you think so?

    }

    <!-- tab: app.component.html -->
    <dx-html-editor [aiIntegration]="aiIntegration"></dx-html-editor>

##### Vue

    <!-- tab: App.vue -->
    <template>

>     <DxHtmlEditor :ai-integration="aiIntegration" />

Reviewer 2: ```suggestion
        <DxHtmlEditor :ai-integration="aiIntegration" />
```

    </template>

    <script setup lang="ts">
    import { AIIntegration } from 'devextreme-vue/ai-integration';
    const ai = new AIIntegration(provider);
    </script>

##### React

    <!-- tab: App.tsx -->
    import { AIIntegration } from 'devextreme-react/ai-integration';

>     const ii = new AIIntegration(provider);

Reviewer 3: ii?)))
